### DISCUSSION TITLE
No Diagram Insert Button in Visual Mode and Markdown->Visual conversion loses diagrams

### DISCUSSION THREAD

**Discussion by gdevenyi** [requarks/wiki#5023] [Error / Bug Report] [answered]
Thanks for this lovely project.

After just beginning to test things out I discovered the amazing diagram insertion tool in the markdown editor, but I can't find an equivalent button in the Visual editor, and if I convert a page with a diagram from markdown to visual it loses the diagrams.

Edit:
And an update, after converting back, it appears the page still has the diagram, but its no longer in the markdown source.

So, the diagram is on the page, but no longer editable in markdown or visual.

**Comment by NGPixel** [comment, marked as answer]
This is [mentionned in the docs](https://docs.requarks.io/editors#change-editor) and not a bug. It's a limitation of switching formats that are not compatible with each other.
